roles/client_side_tests/tasks/test_e2e.yml

```diff
@@ -1,126 +1,133 @@
----
-- name: "Show the prom_url"
-  ansible.builtin.debug:
-    var: prom_url
+    - name: Get the default-prometheus-htpasswd secret
+      ansible.builtin.shell: 
+        cmd: |
+          oc  get secret default-prometheus-htpasswd -ojson | jq '.data.auth, .data.password' |  sed 's/"//g'
+      register: prom_secret
+    - debug: 
+        var: prom_secret | string
 
-- name: "If there's no url set, then grab it from a file"
-  when: prom_url is not defined
-  block:
-    - name: "Extract URL from stf-connectors template"
+    - name: "Get the prom URL"
       ansible.builtin.shell:
         cmd: |
-          grep interconnect /home/stack/virt/stf-connectors* | sed 's;.*\(-service-telemetry.*\);default-prometheus-proxy\1;'
-      register: cmd_output
-      changed_when: false
-      failed_when: "cmd_output.stdout_lines|length == 0"
-
-    - name: "[tmp] set prom creds"
-      ansible.builtin.set_fact:
-        prom_url: "{{ cmd_output.stdout_lines[0] }}"
+          oc get route/default-prometheus-proxy -ojson | jq '.spec.host' | sed 's/"//g'
+      register: prom_route
 
-    - name: "Show the command output"
-      ansible.builtin.debug:
-        var: cmd_output.stdout_lines
+    - debug:
+        var: prom_route.stdout
 
-    - name: "Add info for the the user"
-      ansible.builtin.debug:
-        msg: The URL is {{ cmd_output.stdout_lines }}
+    - name: "Get the prom creds from the secret"
+      ansible.builtin.set_fact:
+        prom_user_decoded: "{{ ( prom_secret.stdout_lines[0] | b64decode ) }}"
+        prom_pass: "{{ prom_secret.stdout_lines[1] | b64decode }}"
+        prom_url: "{{ prom_route.stdout }}"
+    
+    - debug: 
+        var: prom_user
 
-- name: "Add debug info to check the prom_url value"
-  ansible.builtin.debug:
-    msg: The URL is {{ prom_url }}
+    - name: Fetch user
+      ansible.builtin.set_fact:
+        prom_user: "{{ prom_user_decoded.split(':')[0] }}"
```

Comment:
It should be possible to combine this with the ``set_fact`` task above and avoid using a ``prom_user_decoded`` var.

Reply:
We need this because of the split. IR uses old version of ansible that fails on split.